### PR TITLE
Add CRI validation test into presubmit test.

### PR DESCRIPTION
**Pull request by Random-Liu**
Add CRI validation test as part of presubmit test.

Signed-off-by: Lantao Liu <lantaol@google.com>

### CODE REVIEW

File: hack/test-cri.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  echo "GOPATH/bin is not in path"
+  exit 1
+fi
+
+CRITEST=critest
+CRITEST_PKG=github.com/kubernetes-incubator/cri-tools
+CRICONTAINERD_SOCK=/var/run/cri-containerd.sock
+
+# Install critest
+if [ ! -x "$(command -v ${CRITEST})" ]; then
```
Comment (mikebrow) [inline hack/test-cri.sh]: fails if you have the wrong version of critest previously installed as critest is not installed with deps to ensure the correct version.

For example: I got 30 of these: 
```
STEP: create Privileged podSandbox
E0626 07:59:02.698915    7909 remote_runtime.go:91] RunPodSandbox from runtime service failed: rpc error: code = 12 desc = unknown service v1alpha1.RuntimeService
```

File: hack/test-cri.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  go get -d ${CRITEST_PKG}/... 
+  cd ${GOPATH}/src/${CRITEST_PKG}
+  git checkout ${CRITEST_VERSION}
```
Comment (mikebrow) [inline hack/test-cri.sh]: The go get -d did not fetch the commit you listed:
I received this error:
```fatal: reference is not a tree: 2d59bc793e71ee324895ca422e87a863435aa8cd```
I resolved it with:
```sudo git fetch upstream master --tags```

File: hack/install-deps.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+EOF'
+
+# Install containerd
+go get -d ${CONTAINERD_PKG}/...
+cd ${GOPATH}/src/${CONTAINERD_PKG}
+git checkout ${CONTAINERD_VERSION}
+make
+sudo make install
+which containerd
+which containerd-shim
```
Comment (mikebrow) [inline hack/install-deps.sh]: Install cri-tools, since it is a dependency of test-cri?

**Comment by mikebrow** [comment]
I've reproduced the travis error on my ubuntu 14.04lts server (vm image)

appears to be failing to create our sandbox pause container in containerd/shim showing an error from the taskService.Create() call.  

I spun up gcr.io/google_containers/pause:3.0 in runc and that works fine so it's not anything to do with running the image on trusty.. Must be either our spec or some bug / contention in containerd.

**Comment by Random-Liu** [comment]
@mikebrow I had similar problem before on my desktop, which is a trusty based OS. Is overlay enabled on your trusty node?

**Comment by mikebrow** [comment]
It is now.. no difference in the result though.

File: Makefile [hunk 4]
```diff
@@ -71,7 +73,10 @@ cri-containerd: check-gopath
 	   $(PROJECT)/cmd/cri-containerd
 
 test:
-	go test -timeout=10m -v -race ./pkg/... $(BUILD_TAGS)
+	go test -timeout=10m -race ./pkg/... $(BUILD_TAGS)
+
+test-cri:
```
Comment (Random-Liu) [inline Makefile]: Done

File: hack/install-deps.sh [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+go get -d ${CNI_PKG}/...
+cd ${GOPATH}/src/${CNI_PKG}
+git checkout ${CNI_VERSION}
+./build
+sudo mkdir -p ${CNI_DIR}
+sudo cp -r ./bin ${CNI_DIR}
+sudo mkdir -p ${CNI_CONFIG_DIR}
+sudo bash -c 'cat >'${CNI_CONFIG_DIR}'/10-containerd-bridge.conf <<EOF
+{
+	"cniVersion": "0.2.0",
```
Comment (Random-Liu) [inline hack/install-deps.sh]: Yeah, [cni example](https://github.com/containernetworking/cni#running-the-plugins) and [cri-o](https://github.com/kubernetes-incubator/cri-o/tree/master/contrib/cni) are also using this.

File: hack/install-deps.sh [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+		"subnet": "10.88.0.0/16",
+		"routes": [
+			{ "dst": "0.0.0.0/0" }
+		]
+	}
+}
+EOF'
+sudo bash -c 'cat >'${CNI_CONFIG_DIR}'/99-loopback.conf <<EOF
+{
+	"cniVersion": "0.2.0",
```
Comment (Random-Liu) [inline hack/install-deps.sh]: Same answer.

File: hack/test-cri.sh (hunk 2 above)
Comment (Random-Liu) [inline hack/test-cri.sh]: Added `git fetch --all` to address this issue.

File: hack/test-cri.sh (hunk 1 above)
Comment (Random-Liu) [inline hack/test-cri.sh]: Discussed.

**Comment by Random-Liu** [comment]
@mikebrow Figured out the issue. Ubuntu trusty and debian jessie are using old `libseccomp2` version, which is too old for runc. [A backported version](https://github.com/opencontainers/runc/blob/master/Dockerfile#L4) is required.

File: pkg/server/container_create_test.go [hunk 7]
```diff
@@ -467,7 +467,6 @@ func TestCreateContainer(t *testing.T) {
 		imageMetadataErr   bool
 		prepareSnapshotErr error
 		createRootDirErr   error
-		createMetadataErr  bool
```
Comment (mikebrow) [inline pkg/server/container_create_test.go]: already in another closed PR... need to rebase.

Comment (Random-Liu) [inline pkg/server/container_create_test.go]: Done.

**Comment by Random-Liu** [comment]
Apply LGTM based on https://github.com/kubernetes-incubator/cri-containerd/pull/93#pullrequestreview-46918052.

**Pull request merged**